File: frontend/types/currency-rate.ts
/**
 * Currency rate types matching backend schemas.
 */

export interface CurrencyRate {
  id: string;
  currency_code: string;
  rate_to_usd: number;
}

export interface CurrencyRateCreate {
  currency_code: string;
  rate_to_usd: number;
}

export interface CurrencyRateUpdate {
  rate_to_usd: number;
}

export type CurrencyRateResponse = CurrencyRate;

export interface CurrencyRateListResponse {
  items: CurrencyRateResponse[];
  total: number;
}

export interface CurrencyRatesImportResponse {
  rates_date: string;
  updated_codes: string[];
  skipped_not_in_feed: string[];
}

File: backend/app/schemas/currency_rate.py
"""
Currency rate Pydantic schemas for request/response validation.
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID


class CurrencyRateBase(BaseModel):
    """Base schema for a currency rate."""
    currency_code: str = Field(..., min_length=3, max_length=3, description="ISO 4217 currency code (e.g., USD, EUR)")
    rate_to_usd: float = Field(..., gt=0, description="Conversion rate: how many units of this currency = 1 USD")


class CurrencyRateCreate(CurrencyRateBase):
    """Create schema for a currency rate."""
    pass


class CurrencyRateUpdate(BaseModel):
    """Update schema for a currency rate."""
    rate_to_usd: float = Field(..., gt=0, description="Conversion rate: how many units of this currency = 1 USD")


class CurrencyRateResponse(CurrencyRateBase):
    """Response schema for a currency rate."""
    id: UUID

    class Config:
        from_attributes = True


class CurrencyRateListResponse(BaseModel):
    """Schema for currency rate list response."""
    items: List[CurrencyRateResponse]
    total: int


